src/librustc_codegen_ssa/back/linker.rs

```diff
@@ -478,13 +478,14 @@ impl<'a> Linker for GccLinker<'a> {
     }
 
     fn debuginfo(&mut self, strip: Strip) {
+        // MacOS linker does not support longhand arguments --strip-debug and --strip-all
         match strip {
             Strip::None => {}
             Strip::Debuginfo => {
-                self.linker_arg("--strip-debug");
+                self.linker_arg("-s");
```

Comment:
`-s` means `--strip-all`, and `-S` means `--strip-debug` in `ld` (https://sourceware.org/binutils/docs/ld/Options.html).
Is the order actually switched in MacOS linker, or it's a mistake?

Reply:
Agh, it's a mistake... and I triple-checked it too!!